### PR TITLE
GH-75 Added a note regarding the proposer use of php serve.

### PR DESCRIPTION
**Pull request by keithmifsud**
@danielepolencic Please read my questions on #GH-75.

### CODE REVIEW

**Comment by danielepolencic** [comment]
https://github.com/learnk8s/learnk8s.io/blame/52758fba3f2506bf22f6ed2ae008e80f41c970f5/_blog/deploying-laravel-to-kubernetes/index.md#L87

It should be `Dockerfile`. There's no capital `F`.

https://github.com/learnk8s/learnk8s.io/blame/52758fba3f2506bf22f6ed2ae008e80f41c970f5/_blog/deploying-laravel-to-kubernetes/index.md#L74

We should remove `RUN php artisan key:generate` from that Dockerfile. The key should be injected as an env variable. Also, we should use a web server rather than `php artisan serve`.

The proposal is to implement something like this: https://github.com/learnk8s/laravel-kubernetes-demo/pull/1

The above change requires few tweaks.

Deploy to k8s with:

```bash
kubectl run laravel-kubernetes-demo \
  --image=yourname/laravel-kubernetes-demo \
  --port=80 \
  --image-pull-policy=IfNotPresent \
  --env=APP_KEY=base64:cUPmwHx4LXa4Z25HhzFiWCf7TlQmSqnt98pnuiHmzgY=
```

Expose with:

```bash
kubectl expose deployment laravel-kubernetes-demo --type=NodePort --port=80
```

The above should help fix most of the issues that were raised.

File: _blog/deploying-laravel-to-kubernetes/index.md [hunk 1]
```diff
@@ -69,26 +69,50 @@ To follow with this demonstration, you will need the following installed on your
...
+  -p 8080:80 \
+  -e APP_KEY=base64:cUPmwHx4LXa4Z25HhzFiWCf7TlQmSqnt98pnuiHmzgY= \
+  laravel-kubernetes-demo
+```
+
+And the application should be available on `http://localhost:8080`.
+
+With this setup, the container is generic and the `APP_KEY` is not hardcoded or shared.
+
+__Building the image within Minikube__
```
[the remark this answers is not in the document]
Comment (danielepolencic) [reply _blog/deploying-laravel-to-kubernetes/index.md]: The first time you built the image, it was against your local Docker daemon. The second time it's built using the remote Docker daemon in minikube. You can check that yourself: try and build the image and do `docker images`. Connect to minikube with `minikube docker-env` and type `docker images`. The image is not there! Indeed you're dealing with two different daemons (local and remote).

**Comment by danielepolencic** [comment]
As I reader, the assumption is that the article is correct. While I think it's a good idea to display that it is changed since publication, I'd like to look at those articles as evergreen. I don't want to publish and forget about them, I'd like it to be up to date.

**Comment by danielepolencic** [comment]
I'm more and more convinced that we should update the title of the blog post again 😄 

The title of the blog doesn't really make justice to the content and is not addressing one of our new goals in the blog post's checklist:

> [There's a title, and the title is not about the tool, but what you can do with the tool](https://twitter.com/copyconstruct/status/1006599257976008704).

Current title: "How to deploy Laravel to Kubernetes"

Proposed titles:

- Kubernetes: the easiest way to deploy and scale Laravel 5 projects
- Kubernetes: deploy Laravel the easy way
- How to deploy and scale Laravel projects on the same server with Kubernetes (requires content tweaking)
- Deploying and organising Laravel projects on the same server with Kubernetes
- Deploying and organising Laravel projects on the same VPS with Kubernetes
- Deploying, scaling and organising Laravel projects on the same VPS with Kubernetes
- Zero conf deploying and scaling Laravel applications with Kubernetes

**Comment by keithmifsud** [comment]
Hehe, okay, your stubbornness makes sense 😃 

If we change the heading to _Kubernetes: deploy Laravel the easy way_, will we have to change the URL too? How would this affect SEO?

**Comment by danielepolencic** [comment]
We need to add a redirect: https://github.com/jekyll/jekyll-redirect-from

**Comment by keithmifsud** [comment]
Ok. I'll set up an issue for the install. Change the title and the path, test and then we can hopefully merge and share on medium's publications 😄 All this will be done tomorrow.

**Comment by keithmifsud** [comment]
- [ ] Waiting for #122 / #121 Once these are closed, 

- [ ] Update title and URL

- [ ] add CTA as per #119 / learnk8s/team#38

**Comment by danielepolencic** [comment]
I'm confident you committed the right changes.
I suspect the order in which Jekyll loads the plugins does matter.
Have a look here: https://github.com/learnk8s/learnk8s.io/pull/136/files#diff-aeb42283af8ef8e9da40ededd3ae2ab2R38
The redirect plugin should be loaded last.

**Comment by keithmifsud** [comment]
The order of plugins should not affect this. The two other plugins are not hooked to the same event and are called after the output is buffered for printing. In any case, the plugin is loaded last on both the Gem file and the config. https://github.com/learnk8s/learnk8s.io/blob/GH-75/_config.yml https://github.com/learnk8s/learnk8s.io/blob/GH-75/Gemfile

@danielepolencic This seems to be working on Netlify. Can you confirm that? If you don't mind I suggest that you test it on Netlify and if possible on your system. I think the issue is only with Windows.

**Comment by keithmifsud** [comment]
@danielepolencic I also included the CTAs but went ahead and merged the PR.

**Comment by danielepolencic** [comment]
great!

**Pull request merged**